runtime/src/accounts_db.rs

```diff
@@ -3065,56 +3107,148 @@ impl AccountsDb {
         recycle_stores_write_elapsed.as_us()
     }
 
+    // Purges every slot in `removed_slots` from both the cache and storage. This includes
+    // entries in the accounts index, cache entries, and any backing storage entries.
```

Comment:
nit: Thanks for adding comments! I can't overstate how helpful they are. Can you make these doc (`///`) comments? Then they get rendered with the docs and in some editors too.